pkg/clusteragent/clusterchecks/dispatcher_main.go

```diff
@@ -48,11 +54,17 @@ func (d *dispatcher) add(config integration.Config) {
 	if !config.ClusterCheck {
 		return // Ignore non cluster-check configs
 	}
-	log.Debugf("dispatching configuration %s:%s", config.Name, config.Digest())
 
-	// TODO: add dispatching logic
-	hostname, _ := util.GetHostname()
-	d.addConfig(config, hostname)
+	target := d.getLeastBusyNode()
```

Comment:
What do you think about using a queue to store the pending configs to dispatch instead of scheduling them to the dummy node ?
This can be for later, just wanted your thoughts on this ?

(still trying to figure out if the dummy node can introduce latency on the scheduling if we wait for it to be stale)

Reply:
That would be cleaner indeed, although it's basically a safety net if we loose all agents than a common code path.
Max latency DCA-side will be by 30/2 = 15 seconds by default. Add 10 seconds for the node-agent pool frequency

Reply:
Implemented this: dangling configs are now stored in a slice, that is retried every 15 seconds: https://github.com/DataDog/datadog-agent/pull/2392/files#diff-62d6072bb2f0d7d53f6f4299e5395fe9R30